test/v1/models/NinoApplicationSpec.scala

```diff
@@ -0,0 +1,176 @@
+      "return true" when {
+        "the NINO passes validation" in {
+          validateNino(validNino) shouldBe true
+        }
+      }
+      "return false" when {
+        "the NINO fails validation" in {
+          validateNino(invalidNino) shouldBe false
+        }
+      }
+    }
+
+    "using the officeNumber regex" should {
+      "return true" when {
+        "the number passes validation with maximum length" in {
+          validateOffice(validOfficeNumberMax) shouldBe true
+        }
+        "the number passes validation when in between length limits" in {
+          validateOffice(validOfficeNumberMid) shouldBe true
+        }
+        "the number passes validation with minimum length" in {
+          validateOffice(validOfficeNumberMin) shouldBe true
+        }
+      }
+      "return false" when {
+        "the number fails validation due to invalid characters" in {
+          validateOffice(invalidOfficeNumberCharacters) shouldBe false
+        }
+        "the number fails validation due to length" in {
+          validateOffice(invalidOfficeNumberLength) shouldBe false
+        }
+        "the number fails validation due to having zero length" in {
+          validateOffice(invalidOfficeNumberNoLength) shouldBe false
+        }
+      }
+    }
+
+    "using the contactNumber regex" should {
+      "return true" when {
+        "the number passes validation with maximum length" in {
+          validateContact(validContactNumberMax) shouldBe true
+        }
+        "the number passes validation when in between the length limits" in {
+          validateContact(validContactNumberMid) shouldBe true
+        }
+        "the number passes validation with minimum length" in {
+          validateContact(validContactNumberMin) shouldBe true
+        }
+      }
+      "return false" when {
+        "the number fails validation due to invalid characters" in {
+          validateContact(invalidContactNumberCharacters) shouldBe false
+        }
+        "the number fails validation due to length" in {
+          validateContact(invalidContactNumberLength) shouldBe false
+        }
+        "the number fails validation due to having zero length" in {
+          validateContact(invalidContactNumberNoLength) shouldBe false
+        }
+      }
+    }
+  }
+
+  ".validateCountry" should {
+    "return true" when {
+      "the country value is at the minimum" in {
+        NinoApplication.validateCountry(0) shouldBe true
+      }
+      "the country value is at the maximum" in {
+        NinoApplication.validateCountry(286) shouldBe true
+      }
+      "the country value is within range" in {
+        NinoApplication.validateCountry(143) shouldBe true
+      }
+    }
+    "return false" when {
+      "the number is below the minimum" in {
+        NinoApplication.validateCountry(-1) shouldBe false
+      }
+      "the number is above the maximum" in {
+        NinoApplication.validateCountry(300) shouldBe false
```

Comment:
It's always worth going just above the max range with these tests. So I'd use 287 instead of 300

Reply:
Please can you look at the comment above